src/uwtools/config/formats/yaml.py

```diff
@@ -43,7 +43,8 @@ def __repr__(self) -> str:
         """
         The string representation of a YAMLConfig object.
         """
-        return yaml.dump(self.data)
+        yaml.add_representer(TaggedString, TaggedString.represent)
+        return yaml.dump(self.data, default_flow_style=False).strip()
```

Comment:
Add a "representer" function for `TaggedString` objects, representing them as `!tag "value"`.